pkg/skaffold/kubernetes/portforward/resource_forwarder_test.go

```diff
@@ -206,51 +206,87 @@ func TestUserDefinedResources(t *testing.T) {
 	svc := &latest.PortForwardResource{
 		Type:      constants.Service,
 		Name:      "svc1",
-		Namespace: "default",
+		Namespace: "test",
 		Port:      8080,
 	}
 
-	pod := &latest.PortForwardResource{
-		Type:      constants.Pod,
-		Name:      "pod",
-		Namespace: "default",
-		Port:      9000,
-	}
-
-	expected := map[string]*portForwardEntry{
-		"service-svc1-default-8080": {
-			resource:  *svc,
-			localPort: 8080,
+	tests := []struct {
+		description       string
+		userResources     []*latest.PortForwardResource
+		namespaces        []string
+		expectedResources []string
+	}{
+		{
+			description: "one service and one user defined pod",
+			userResources: []*latest.PortForwardResource{
+				{Type: constants.Pod, Name: "pod", Namespace: "some", Port: 9000},
+			},
+			namespaces: []string{"test"},
+			expectedResources: []string{
+				"service-svc1-test-8080",
+				"pod-pod-some-9000",
+			},
 		},
-		"pod-pod-default-9000": {
-			resource:  *pod,
-			localPort: 9000,
+		{
+			userResources: []*latest.PortForwardResource{
+				{Type: constants.Pod, Name: "pod", Port: 9000},
+			},
+			namespaces: []string{"test"},
+			expectedResources: []string{
+				"service-svc1-test-8080",
+				"pod-pod-test-9000",
+			},
+		},
+		{
+			userResources: []*latest.PortForwardResource{
+				{Type: constants.Pod, Name: "pod", Port: 9000},
+			},
+			namespaces: []string{"test", "some"},
+			expectedResources: []string{
+				"service-svc1-test-8080",
+			},
+		},
+		{
+			userResources: []*latest.PortForwardResource{
+				{Type: constants.Pod, Name: "pod", Port: 9000},
+				{Type: constants.Pod, Name: "pod", Namespace: "some", Port: 9001},
+			},
+			namespaces: []string{"test", "some"},
+			expectedResources: []string{
+				"service-svc1-test-8080",
+				"pod-pod-some-9001",
+			},
 		},
 	}
 
-	testutil.Run(t, "one service and one user defined pod", func(t *testutil.T) {
-		event.InitializeState(latest.Pipeline{}, "", true, true, true)
-		t.Override(&retrieveAvailablePort, mockRetrieveAvailablePort("127.0.0.1", map[int]struct{}{}, []int{8080, 9000}))
-		t.Override(&retrieveServices, func(string, []string) ([]*latest.PortForwardResource, error) {
-			return []*latest.PortForwardResource{svc}, nil
-		})
+	for _, test := range tests {
+		testutil.Run(t, test.description, func(t *testutil.T) {
+			event.InitializeState(latest.Pipeline{}, "", true, true, true)
+			t.Override(&retrieveAvailablePort, mockRetrieveAvailablePort("127.0.0.1", map[int]struct{}{}, []int{8080, 9000}))
+			t.Override(&retrieveServices, func(string, []string) ([]*latest.PortForwardResource, error) {
+				return []*latest.PortForwardResource{svc}, nil
+			})
 
-		fakeForwarder := newTestForwarder()
-		entryManager := NewEntryManager(ioutil.Discard, fakeForwarder)
+			fakeForwarder := newTestForwarder()
```

Comment:
unintentional tab.

Reply:
Sorry I'm not sure about this. I have shifted the whole nested function body because of the loop above

Reply:
oops. sorry i missed you ended up shifting.I thought the IDE replaced spaces by tabs.